src/openxr/OpenXRApi.cpp

```diff
@@ -403,24 +356,53 @@ OpenXRApi::OpenXRApi() {
 	result = xrCreateInstance(&instanceCreateInfo, &instance);
 	if (!xr_result(result, "Failed to create XR instance.")) {
 		free(enabledExtensions);
-		return;
+		return false;
 	}
 	free(enabledExtensions);
 
-	if (!initialiseExtensions()) {
-		return;
+	return true;
+}
+
+bool OpenXRApi::initialiseExtensions() {
+	XrResult result;
+
+	// Maybe we should remove the error checking here, if the extension is not supported, we won't be doing anything with this.
+	Godot::print("OpenXR initialiseExtensions");
+	if (hand_tracking_ext) {
+		result = xrGetInstanceProcAddr(instance, "xrCreateHandTrackerEXT", (PFN_xrVoidFunction *)&xrCreateHandTrackerEXT_ptr);
+		if (!xr_result(result, "Failed to obtain xrCreateHandTrackerEXT function pointer")) {
+			return false;
+		}
+
+		result = xrGetInstanceProcAddr(instance, "xrDestroyHandTrackerEXT", (PFN_xrVoidFunction *)&xrDestroyHandTrackerEXT_ptr);
+		if (!xr_result(result, "Failed to obtain xrDestroyHandTrackerEXT function pointer")) {
+			return false;
+		}
+
+		result = xrGetInstanceProcAddr(instance, "xrLocateHandJointsEXT", (PFN_xrVoidFunction *)&xrLocateHandJointsEXT_ptr);
+		if (!xr_result(result, "Failed to obtain xrLocateHandJointsEXT function pointer")) {
+			return false;
+		}
 	}
 
+	return true;
+}
+
+bool OpenXRApi::initialiseSession() {
+	XrResult result;
+
+	Godot::print("OpenXR initialiseSession");
+
 	// TODO: Support AR?
 	XrSystemGetInfo systemGetInfo = {
 		.type = XR_TYPE_SYSTEM_GET_INFO,
 		.next = NULL,
-		.formFactor = XR_FORM_FACTOR_HEAD_MOUNTED_DISPLAY,
+		.formFactor = form_factor,
 	};
 
 	result = xrGetSystem(instance, &systemGetInfo, &systemId);
-	if (!xr_result(result, "Failed to get system for HMD form factor.")) {
-		return;
+	if (!xr_result(result, "Failed to get system for our form factor.")) {
```

Comment:
By the way: Unlike vulkan, there is a reflection header maintained upstream that can be used to stringify enums.
https://github.com/KhronosGroup/OpenXR-SDK/blob/master/include/openxr/openxr_reflection.h